### PR TITLE
[ticket/8920] Add a MCP PM reports view permission

### PR DESCRIPTION
**Pull request by Zoddo**
Anybody who has the permission to read reports in one forum can read PM reports.

Solving this problem by adding a new permission to read PM reports.

https://tracker.phpbb.com/browse/PHPBB3-8920

### CODE REVIEW

File: phpBB/phpbb/db/migration/data/v31x/m_pm_report.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+{
+	static public function depends_on()
+	{
+		return array('\phpbb\db\migration\data\v31x\v316rc1');
+	}
+
+	public function update_data()
+	{
+		return array(
+			array('permission.add', array('m_pm_report', true, 'm_report')),
```
Comment (Nicofuma) [inline phpBB/phpbb/db/migration/data/v31x/m_pm_report.php]: you should add the permission to the MODERATOR rule too

Comment (Zoddo) [inline phpBB/phpbb/db/migration/data/v31x/m_pm_report.php]: The permission is copied from `m_report`.

Comment (Nicofuma) [inline phpBB/phpbb/db/migration/data/v31x/m_pm_report.php]: didn't see... that's good so

**Comment by Nicofuma** [comment]
after running the migration, the permission was set to no for the full moderator on my test board...

**Comment by rxu** [comment]
It probably needs the ```permission.permission_set``` as well to apply new permission for groups/roles.

**Comment by Zoddo** [comment]
@rxu : So, a bug with the "permission copy" function?

**Comment by rxu** [comment]
Tested locally like that
```php
$permission = new \phpbb\db\migration\tool\permission($db, $cache, $auth, $phpbb_root_path, $phpEx);
$permission->add('m_pm_report', true, 'm_report');
```

Checked the permission for the Full moderator and it's been set to ```Yes```.

@Nicofuma any chance you have set the ```m_report``` permission to ```No``` for the full moderator so it's just copied to the new permission?

**Comment by Zoddo** [comment]
Hi,

Any news?

**Comment by marc1706** [comment]
Looks good. Can't reproduce the issue mentioned by @Nicofuma

**Pull request merged**